### PR TITLE
Implement message passing from website

### LINKED ISSUE #25: Implement function to receive messages through the background
Implement functions that receives message when sending a message in a background web page.

### PR DESCRIPTION
**Pull request by HyeockJinKim**
Closes #25

### CODE REVIEW

File: background/contentScript.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  if (event.source !== window)
+    return;
+
+  if (event.data.from && (event.data.from === 'page')) {
+    chrome.runtime.sendMessage({
+      from: 'content',
+      data: event.data,
+    });
+  }
+}, false);
```
Comment (HyeockJinKim) [inline background/contentScript.js:12]: A function that runs in the background of the extension and detects `postMessage`.

File: manifest.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "0Auth Extension",
   "version": "1.0",
   "description": "Chrome Extension for 0Auth",
-  "permissions": ["declarativeContent", "storage"],
+  "permissions": ["activeTab", "declarativeContent", "storage"],
```
Comment (HyeockJinKim) [inline manifest.json:5]: permission for contentscript.

File: manifest.json [hunk 3]
```diff
@@ -13,6 +13,13 @@
   "icons": {
     "128": "public/icon.png"
   },
+  "content_scripts": [
+    {
+      "matches": ["<all_urls>"],
+      "all_frames": true,
+      "js": ["background/contentScript.js"]
+    }
+  ],
```
Comment (HyeockJinKim) [inline manifest.json:22]: contentscript.js run in `<all_site>`

File: src/App.tsx [hunk 4]
```diff
@@ -74,26 +74,19 @@ export default function App() {
...
-          title={dummyData.title}
-          currentInfos={dummyData.currentInfos}
-          signature={dummyData.signature}/>
-        <LoginPage path='/login'/>
-        <SettingPage path='/settings'/>
-        <RegisterPage
-          path='/register'
-          favicon={dummyData.favicon}
-          title={dummyData.title}
-          dynamicFormInputs={dummyData.formInputs}/>
```
Comment (HyeockJinKim) [inline src/App.tsx:90]: Moved props values to mobx store.

File: src/components/authenticate/index.tsx [hunk 5]
```diff
@@ -9,17 +9,16 @@ import {
 
 import {WebsiteInfo, BaseButton} from '../common';
 import {PrivatePage} from '../page';
+import {currentStore} from "../../stores";
 
+// TODO: Move to 0auth library!
 type RequestedInfo = {
   label: string;
   required: boolean;
 }
```
Comment (HyeockJinKim) [inline src/components/authenticate/index.tsx:18]: This type must be shared in the `0auth` library.

File: src/components/page/public/index.tsx [hunk 6]
```diff
@@ -21,13 +23,37 @@ const useStyles = makeStyles(() => ({
...
+  const history = useHistory();
+  const register = (form: DynamicFormInput[]) => {
+    history.push('/register');
+    currentStore.setRegister(form);
+  };
+  const auth = (properties: Property[] | undefined, sign: Signature | undefined) => {
+    history.push('/auth');
+    if (properties !== undefined) {
+      currentStore.setAuth(properties, sign);
+    }
```
Comment (HyeockJinKim) [inline src/components/page/public/index.tsx:37]: move page and set data.

File: src/components/page/public/index.tsx [hunk 7]
```diff
@@ -21,13 +23,37 @@ const useStyles = makeStyles(() => ({
...
+          register(msg.data.form as DynamicFormInput[]);
+          break;
+        case 'AUTH':
+          auth(msg.data.properties as Property[], msg.data.sign as Signature);
+          break;
+        default:
+          throw Error('Unreachable Code in onMessage');
+      }
+    });
+  }
```
Comment (HyeockJinKim) [inline src/components/page/public/index.tsx:52]: Add message listener for message passed by website.

File: src/stores/current/index.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import {makeAutoObservable} from 'mobx';
+import {Property, Signature} from "@0auth/message";
+import {DynamicFormInput} from "../../components/register";
+
+class Current {
```
Comment (HyeockJinKim) [inline src/stores/current/index.ts:5]: A mobx store that stores values to be used in the current user page.

**Review by HyeockJinKim — COMMENTED** [review]
self review.

File: src/components/login/index.tsx [hunk 9]
```diff
@@ -61,20 +61,27 @@ const LoginPage = observer(({path}: LoginPageProps) => {
...
       </Typography>
       <form className={classes.form}>
-        <TextField className={classes.input} value={loginStore.password} onChange={e => loginStore.setPassword(e.target.value)} type="password" label="Password"/>
+        <TextField className={classes.input} onKeyPress={e => enter(e)} value={loginStore.password} onChange={e => loginStore.setPassword(e.target.value)} type="password" label="Password"/>
         <BaseButton icon={<LockOpenIcon/>} onClick={login}>
           Start 0auth
         </BaseButton>
       </form>
-      <Button className={classes.setPass}>Set your password</Button>
+      <Button onClick={e => history.push('/settings')} className={classes.setPass}>Set your password</Button>
```
Comment (anthonyminyungi) [inline src/components/login/index.tsx:84]: You don't have to change them, but I want to let you know a better way.

```jsx
import React, { useCallback } from 'react';

...

  const keyPressHandler = useCallback(e: React.KeyboardEvent<HTMLDivElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      login();
    }  },[/*state for update*/])
    // but, not have to be useCallback.

...

  return (
    <TextField onKeyPress={keyPressHandler} /*...*/  />
    ...
  )
```

If you write the event handler this way, you don't have to write the anonymous function in the handler 
like `{e => enter(e)}`.

**Review by anthonyminyungi — APPROVED** [review]
Cool. LGTM.

**Pull request merged**